src/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -840,7 +837,8 @@ internal sealed class AwaitableSocketAsyncEventArgs : SocketAsyncEventArgs, IVal
 
             public bool WrapExceptionsInIOExceptions { get; set; }
 
-            public bool Reserve() => Interlocked.CompareExchange(ref _continuation, null, s_availableSentinel) == s_availableSentinel;
+            public bool Reserve() =>
+                ReferenceEquals(Interlocked.CompareExchange(ref _continuation, null, s_availableSentinel), s_availableSentinel);
```

Comment:
ReferenceEquals is a tad faster and is all we care about.

Reply:
Delegate.op_Equality does show up in my perf traces (not huge, but like 0.5%), so this is definitely a win.